### ISSUE TITLE
Feature request: Indent by specificity

### ISSUE THREAD

**Issue by jakelauer** [stylelint/stylelint#2443]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A feature that requires indentation by specificity, such that a more specific rule (`#my-element .child`) that follows a less specific rule (`#my-element`) is indented.

> Which rule, if any, is this issue related to?

`indentation`

> What CSS is needed to reproduce this issue?

This should trigger a warning:

```css
#my-element
{
    color: red;
}

#my-element .child
{
    float: left;
}
```

This would not trigger a warning:

```css
#my-element
{
    color: red;
}

    #my-element .child
    {
        float: left;
    }
```

> What stylelint configuration is needed to reproduce this issue?

N/A

> Which version of stylelint are you using?

7.9.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS CLI via stylefmt 

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

Warning that shows up when indentation rule is not followed (if the rule is enabled)

> What actually happened (e.g. what warnings or errors you are getting)?

This rule does not yet exist

**Comment by davidtheclark** [comment]
Thanks for the suggestion, @jakelauer. I think this rule idea would make a great plugin, with some enthusiastic users. However, I think the rule would be too specialized, complicated, and error-prone for us to include it as a core rule — at least not before it's been vetted as a plugin, first.

Please have look at [our guide for writing plugins](https://stylelint.io/developer-guide/plugins/) to help you get started.
